src/components/Gallery/ImageGallery.tsx

```diff
@@ -29,11 +30,39 @@ export class Gallery extends Component<GalleryProps, GalleryState> {
     };
   }
 
+  /**
+   * Fetches an Object of asset metadata, like width and height attributes.
+   */
+  getAssetMetadata = async (assetURL: string) => {
+    const response = await fetch(`${assetURL.split('?')[0]}?fm=json`);
+    return await response.json();
+  };
+
   handleClick = (selectedAsset: AssetProps) => this.setState({ selectedAsset });
 
-  handleSubmit = () => {
+  handleSubmit = async () => {
+    if (!this.state.selectedAsset?.src) {
+      return;
+    }
+
+    // add metadata to selectedAsset attributes
+    const metadata = await this.getAssetMetadata(this.state.selectedAsset.src);
+    const selectedAsset = { ...this.state.selectedAsset };
+
+    if (!selectedAsset.attributes.media_width) {
+      selectedAsset.attributes.media_width = metadata.PixelWidth;
+    }
+
+    if (!selectedAsset.attributes.media_height) {
+      selectedAsset.attributes.media_hight = metadata.PixelHeight;
```

Comment:
typo (blocking)
```suggestion
      selectedAsset.attributes.media_height = metadata.PixelHeight;
```

Reply:
comment:

Shoot nice catch 😬 ty. 

That `attributes` types TODO really coming back to bite us here.